src/DIRAC/Core/Utilities/Plotting/FileCoding.py

```diff
@@ -46,7 +46,7 @@ def extractRequestFromFileId(fileId):
   if compressType == 'Z':
     gLogger.info("Compressed request, uncompressing")
     try:
-      stub = base64.urlsafe_b64decode(stub)
+      stub = base64.urlsafe_b64decode(stub.encode())
```

Comment:
This was surprising to me but it turns out Python 2 can't handle unicode base64 data, maybe add a comment so we remember it's a Python 2 quirk.

```python
christopherburr@maclhcb60 ~> python2.7
Python 2.7.16 (default, Jan 27 2020, 04:46:15)
[GCC 4.2.1 Compatible Apple LLVM 10.0.1 (clang-1001.0.37.14)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import base64
>>> base64.urlsafe_b64decode("SGVsbG8gd29ybGQK")
'Hello world\n'
>>> base64.urlsafe_b64decode(b"SGVsbG8gd29ybGQK")
'Hello world\n'
>>> base64.urlsafe_b64decode(u"SGVsbG8gd29ybGQK")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/base64.py", line 119, in urlsafe_b64decode
    return b64decode(s.translate(_urlsafe_decode_translation))
TypeError: character mapping must return integer, None or unicode

christopherburr@maclhcb60 ~> python3
Python 3.9.1 (default, Dec 28 2020, 11:22:14)
[Clang 11.0.0 (clang-1100.0.33.17)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import base64
>>> base64.urlsafe_b64decode("SGVsbG8gd29ybGQK")
b'Hello world\n'
>>> base64.urlsafe_b64decode(b"SGVsbG8gd29ybGQK")
b'Hello world\n'
>>> base64.urlsafe_b64decode(u"SGVsbG8gd29ybGQK")
b'Hello world\n'
```

```suggestion
      # Encoding is only required for Python 2 and can be removed for WebAppDIRAC v5
      stub = base64.urlsafe_b64decode(stub.encode())
```

Reply:
The only thing I don't understand is why it should be based on WebAppDIRAC? DIRAC extensions depend on DIRAC, not the other way around. IMHO this fix can be removed along with other code related to PY2 compatibility in DIRAC version that stops supporting PY2.

Reply:
You’re right. I thought this was in WebAppDIRAC not vanilla DIRAC. 🤦